structs.go

```diff
@@ -132,7 +133,13 @@ func (s *Struct) FillMap(out map[string]interface{}) {
 			continue
 		}
 
-		out[name] = finalVal
+		if isSubStruct && (tagOpts.Has("flatten")) {
+			for k := range finalVal.(map[string]interface{}) {
+				out[k] = finalVal.(map[string]interface{})[k]
+			}
+		} else {
+			out[name] = finalVal
+		}
```

Comment:
@fatih my english is not good (this could be ok?)
// A tag value with the option of "flatten" used in a struct field is to flatten its fields
// in the output map. Example:
//
//   // The FieldStruct's fields will be flattened into the output map.
//   FieldStruct time.Time `structs:"flatten"`
//